gen3utils/gitops/gitops_validator.py

```diff
@@ -0,0 +1,421 @@
+                [_, str_props] = (
+                    list(filter(None, re.split(r"[\[\]]", item)))
+                    if "[" in item
+                    else [item, None]
+                )
+                if str_props is not None:
+                    props = str_props.split(",")
+                    index_props.extend(
+                        [
+                            p.split(":")[0].strip() if p.find(":") != -1 else p.strip()
+                            for p in props
+                        ]
+                    )
+
+        all_prop_map[index.get("doc_type")] = set(index_props)
+
+    return all_prop_map
+
+
+def _extract_props(props_to_extract):
+    if not props_to_extract:
+        return []
+    return [p["name"] for p in props_to_extract]
+
+
+def validate_against_dictionary(gitops, data_dictionary):
+    """
+    Validates gitops.json configuration against a data dictionary
+    Args:
+        gitops (dict): gitops.json config
+        data_dictionary (str): url of data dictionary
+
+    Returns:
+        ok(bool): whether the validation succeeded.
+
+    """
+
+    _, model = init_dictionary(data_dictionary)
+    schema = model.dictionary.schema
+
+    ok = True
+    graphql = gitops["graphql"]
+    for item in graphql["boardCounts"]:
+        node_count = item["graphql"]
+        # assumes form _{node}_count
+        idx = node_count.rfind("_")
+        node = node_count[1:idx]
+
+        ok = ok and assert_and_log(
+            schema.get(node) is not None,
+            "Node: {} in graphql.boardCounts not found in dictionary".format(node),
+        )
+
+    for item in graphql["chartCounts"]:
+        node_count = item["graphql"]
+        # assumes form _{node}_count
+        idx = node_count.rfind("_")
+        node = node_count[1:idx]
+        ok = ok and assert_and_log(
+            schema.get(node) is not None,
+            "Node: {} in graphql.chartCounts not found in dictionary".format(node),
+        )
+
+    for item in graphql.get("homepageChartNodes", []):
+        node = item["node"]
+        ok = ok and assert_and_log(
+            schema.get(node) is not None,
+            "Node: {} in graphql.homepageChartNodes not found in dictionary".format(
+                node
+            ),
+        )
+
+    return ok
+
+
+def comment_gitops_errors_on_pr(repository, pull_request_number, gitops_errors):
+    token = os.environ["GITHUB_TOKEN"]
+    headers = {"Authorization": "token {}".format(token)}
+
+    repository = repository.strip("/")
+    base_url = "https://api.github.com/repos/{}".format(repository)
+    logger.info("Checking pull request: {} #{}".format(repository, pull_request_number))
+    pr_comments_url = "{}/issues/{}/comments".format(base_url, pull_request_number)
+    contents = ""
+    for error in gitops_errors:
+        contents += "## Gitops-etlMapping : {}".format(error)
+    full_comment = "# {}\n{}".format("gitops.json", contents)
```

Comment:
we could make the comment prettier, explain the contents and instruct to ignore it if following the ETL/portal deployment process. i can help write that. for now we can keep this for testing